MapboxNavigation/RouteController.swift

```diff
@@ -110,9 +107,46 @@ extension RouteController: CLLocationManagerDelegate {
             }
         }
         
+        if routeProgress.currentLegProgress.stepIndex == 0 {
+            let step = routeProgress.currentLegProgress.currentStepProgress.step
+            let stepLocation = CLLocation(latitude: step.coordinates!.first!.latitude, longitude: step.coordinates!.first!.longitude)
+            
+            var currentUserToStartOfRoute = location.distance(from: stepLocation)
+            if currentUserToStartOfRoute > lastUserDistanceToStartOfRoute {
```

Comment:
This seems to calculate the user’s course by storing and comparing the last _n_ distances. Could we achieve the same effect or better by looking at whether `location.course` points toward the route? You could do that by comparing `location.course` to the direction of the closest path to the route line (using `CLLocationCoordinate2D.direction(to:)`).